### DISCUSSION TITLE
How do you create a fixed (perpetual) guarding mission? (like the Archons' one)

### DISCUSSION THREAD

**Discussion by esksp** [endless-sky/endless-sky#10793] [Q&A]
I want to have a "boss" type ship patrolling a system, always, and it has to be unique, so I can't use fleets.

I thought about the Archons guarding specific systems, and tried to copy their code, which seems rather simple at first view.
**The problem is how do I make this pseudo-mission _start_?**
The Archon ones just have a line like "`not "Drak guarding Korath space: active"`", which isn't very explicit, and the Wiki doesn't help (as usual). I didn't find anything else referring to that line in the "/drak" subfolder. It must point to somewhere else.

So, what should I put there for mine? Unlike the Archons, it isn't affected by story, it is just window dressing, it just needs to run all the time, forever (at least till the unlikely event the player kills that ship). 
Here is my glorious attempt at copy/paste... 🙄 

```
mission "Warhip guarding Certified"
	invisible
	landing
	repeat
	to offer
		- ?????? -
	to complete
		never
	npc save
		government "Strike Forces"
		personality heroic vindictive staying frugal
		system Certified
		ship Peacemaker

```

Also, it seems I need to declare the ship in a special way, not just give a ship type ("Peacemaker" is the ship type designation): The Archons seem to refer to an individualized NPC ship declaration but, not having found where this is declared, I don't know how to do that.  :-(  

Thanks for any help...

**Comment by TheGiraffe3** [comment]
Are you using that for `on offer`?
You probably need something a bit different (such as `0 = 0`)

**Reply by esksp** [reply]
> Are you using that for `on offer`?

Sorry, I don't understand your question. I've used the tried & tested method of "copy & paste", copied one of the Archon system guarding missions, and changed the parts I wanted to be different. I want this mission to start automatically and run perpetually.

>  You probably need something a bit different (such as `0 = 0`)

What do you mean? Is that a mission running condition? And if yes, how do I use it?

**Reply by TheGiraffe3** [reply]
Actually, realized: you need an `accept`.
So overall:
```
mission "Warhip guarding Certified"
	invisible
	landing
	repeat
	to offer
		1 = 1
	on offer
		accept
	to complete
		never
	npc save
		government "Strike Forces"
		personality heroic vindictive staying frugal
		system Certified
		ship Peacemaker
```

**Reply by esksp** [reply]
Ow, thanks!

> 		1 = 1

If you have the time, what does this truism actually mean (so I don't die stupid)?...

**Reply by danaris** [reply]
It's basically just a way of saying `true`

**Reply by TheGiraffe3** [reply]
It means the mission will offer.
The game checks if the number one is equal to the number one. If the answer is yes, then the mission can offer.
It's a very funny way to make a mission always offer.

**Reply by ziproot** [reply]
You don’t actually need a `to offer`, because the game defaults that to true. Also, `on offer` should default to accept.

So the actual mission would look like this:
```
mission "Warhip guarding Certified"
	invisible
	landing
	repeat
	to complete
		never
	npc save
		government "Strike Forces"
		personality heroic vindictive staying frugal
		system Certified
		ship Peacemaker
```

**Reply by esksp** [reply]
> the game defaults that to true

That does make sense, thanks!

Except it does not work in the game (no ship). :-(

So something is missing/wrong, probably with the ship's definition?
I noticed that all missions refer to a personalized version of a ship, but I never found where that personalization happens.
Where is (for instance) `ship "Archon" "Kasiliri's Folly"` declared? I'm taking the Archons as example because the "/drak" folder only contains 4 files, so it's easy to sift through! And "Kasiliri's Folly" is declared nowhere!  :-(

**Reply by TheGiraffe3** [reply]
Oh. It's the name of the ship. So you'd need to do 
```
mission "Warship guarding Certified"
	invisible
	landing
	repeat
	to complete
		never
	npc save
		government "Strike Forces"
		personality heroic vindictive staying frugal
		system Certified
		ship Peacemaker "Peacemaker (or ship's name)"
```

**Reply by esksp** [reply]
> 		ship Peacemaker "Peacemaker (or ship's name)"

Thanks, I copied your code and put a name in it (`ship Peacemaker "FSS Hell"`), still no ship appearing...
Apparently this doesn't trigger for some reason. :-(

**Reply by TheGiraffe3** [reply]
And did the one you marked as answer work?
If not, maybe try adding a name there?

**Reply by esksp** [reply]
> And did the one you marked as answer work?

No, actually none worked, in no case there is a ship in that system (and it's a _big_ one, not easy to miss...).
**The mission clearly doesn't trigger** (start), since all active missions appear in the savegames, and this one doesn't appear. 

>  If not, maybe try adding a name there?

Where? If you mean name the ship, I did, as I said above: `ship Peacemaker "FSS Hell"`, no change.

**Reply by warp-core** [reply]
What are you doing to try to get the mission to trigger?
Have you checked the errors file next to the save file folder?

**Reply by esksp** [reply]
> What are you doing to try to get the mission to trigger?

Nothing: I was under the impression this would create an auto-trigger, perpetual mission.
(It doesn't depend on conditions, it just has to run from game start to game end. That ship is a guardian.).

> Have you checked the errors file next to the save file folder?

Hah! I had not, but there are no errors pertaining to that mission.

**Reply by warp-core** [reply]
> > What are you doing to try to get the mission to trigger?
> 
> Nothing: I was under the impression this would create an auto-trigger, perpetual mission. (It doesn't depend on conditions, it just has to run from game start to game end. That ship is a guardian.).

Yeah, but when you launch the game, what are you doing?

**Reply by esksp** [reply]
> Yeah, but when you launch the game, what are you doing?

Nothing so far. Should I do something? And what exactly?

**Reply by TheGiraffe3** [reply]
You need to land to get the mission.

**Reply by esksp** [reply]
> You need to land to get the mission.

🎉 It worked!
Thanks!

**Reply by TheGiraffe3** [reply]
Thank warp-core for pointing it out.

**Reply by esksp** [reply]
Thank you warp-core!... (Even if I didn't understand your allusion)

PS:
Unfortunately GitHub doesn't allow to put several "answer" tags, because usually an answer is teamwork, with several people all contributing a piece. So I just use the "answer" tag to mark the thread as "not needing any more input".

**Reply by esksp** [reply]
LOL! OMG, now they breed like rabbits!  :-D

Apparently **every time I land somewhere, it creates a new instance!**... :-D
There are over 20 of them patrolling that system by now... :-D  :-D  :-D

How can I make sure there is _only one_?

**Reply by warp-core** [reply]
Add:
```
to offer
    not "Warhip guarding Certified: active"
```
To the mission. (Adjust the name of the mission in the condition as appropriate.)

**Reply by esksp** [reply]
Thank you, that worked! 🎉
